### ISSUE TITLE
NetEx and other improvements

### ISSUE THREAD

**Issue by aleks-f** [pocoproject/poco#682]
Based on this [pull](https://github.com/pocoproject/poco/pull/647), there are new [ASN](https://github.com/pocoproject/poco/tree/NetEx/ASN) and [NetEx](https://github.com/pocoproject/poco/tree/NetEx/NetEx) libraries with new SMTP package and ICMP, NTP packages moved from Net library. Also, there is new [HTTPClient](https://github.com/pocoproject/poco/blob/NetEx/Net/include/Poco/Net/HTTPClient.h) class in the Net library for asynchronous (thread-based) HTTP functionality.

**Comment by aleks-f** [comment]
@RangelReale : can you please add some more documentation comments to [ASN1Types.h](https://github.com/pocoproject/poco/blob/NetEx/ASN/include/Poco/ASN/ASN1Types.h)

**Comment by aleks-f** [comment]
@RangelReale : [testPing](https://github.com/pocoproject/poco/blob/NetEx/NetEx/testsuite/src/ICMPClientTest.cpp#L49) fails, even when I comment everything but first two lines; tried Linux and Windows. Can you please look at it?

**Comment by RangelReale** [comment]
I ran testPing on my Windows machine and it ran successfully, but as it is by default raw sockets, so I had to run VS as Administrator.
Strangely sometimes the ping to www.appinf.com times out, maybe it is my network.
I think it appears to be a little too fast to timeout, the Windows ping command seems much slower to run, but never failed in the tests I just did.


Pinging localhost [127.0.0.1] with 48 bytes of data:
-------------------------------------------------------

Reply from 127.0.0.1 bytes=48 time=1ms TTL=128

--- Ping statistics for 127.0.0.1 ---
Packets: Sent=1, Received=1 Lost=0 (0% loss),
Approximate round trip times in milliseconds:
Minimum=1ms, Maximum=1ms, Average=1ms
-----------------------------------------------


Pinging web.appinf.com [50.57.108.29] with 48 bytes of data:
-------------------------------------------------------

Reply from 50.57.108.29 bytes=48 time=218ms TTL=128
Reply from 50.57.108.29 bytes=48 time=171ms TTL=128
Reply from 50.57.108.29 bytes=48 time=265ms TTL=128
Reply from 50.57.108.29 bytes=48 time=281ms TTL=128

--- Ping statistics for 50.57.108.29 ---
Packets: Sent=4, Received=4 Lost=0 (0% loss),
Approximate round trip times in milliseconds:
Minimum=171ms, Maximum=281ms, Average=233ms
-----------------------------------------------


Pinging 192.168.243.1 [192.168.243.1] with 48 bytes of data:
-------------------------------------------------------

192.168.243.1: Request timed out.

--- Ping statistics for 192.168.243.1 ---
Packets: Sent=1, Received=0 Lost=1 (100% loss),
Approximate round trip times in milliseconds:
Minimum=0ms, Maximum=0ms, Average=0ms
-----------------------------------------------


Pinging 10.11.12.13 [10.11.12.13] with 48 bytes of data:
-------------------------------------------------------

10.11.12.13: Request timed out.

--- Ping statistics for 10.11.12.13 ---
Packets: Sent=1, Received=0 Lost=1 (100% loss),
Approximate round trip times in milliseconds:
Minimum=0ms, Maximum=0ms, Average=0ms
-----------------------------------------------
